core/multigrid/pgm.cpp

```diff
@@ -270,20 +279,30 @@ void Pgm<ValueType, IndexType>::communicate(
         send_agg.get_size(), local_agg.get_const_data(),
         gather_idxs.get_const_data(), send_agg.get_data()));
 
+    // There is no index map on the coarse level yet, so map the local indices
+    // to global manually
```

Comment:
```suggestion
    // to global manually on the coarse partition
```

Reply:
Just to be sure. For example, 8 x 8 matrix with two partition and each contains 4. After aggregation, each aggregate to two nodes (i.e. 4 x 4 matrix as global coarse matrix)
first aggregation {0, 1} -> {0, 1}
second aggregation {0, 1} -> {2, 3}
i-th rank maps the local agg to global by local_idx + the coarse matrix offsets by rank (0 for first, 2 for the second in the example). Do I get the function meaning here correctly?

Reply:
I don't really understand your question, so I will try to explain this more generally.
This operation is totally separate from the notion of aggregation. The only relevant part is that there are some local indices (which correspond to the local aggregation indices) and a partition of the global index space these local indices live in.
The local indices are mapped to the corresponding global indices. That's already all. The meaning of the indices, i.e. that they define the aggregation, doesn't really matter here.